### PR TITLE
Add a CommandBarHelper that allows apps to use CommanBar without using a Frame

### PR DESCRIPTION
**Pull request by dr1rrb**
## PR Type
What kind of change does this PR introduce?
 - Feature

## What is the current behavior?
`CommandBar` are tightly coupled to the `Frame` and cannot be used without one.

## What is the new behavior?
We are now able to render a `CommandBar` using a `UINavigationController` not managed by a `Frame`.

## PR Checklist
- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs [hunk 1]
```diff
@@ -1572,7 +1572,7 @@ public bool Equals(WeakReference x, WeakReference y)
 
 			public int GetHashCode(WeakReference obj)
 			{
-				return obj?.GetHashCode() ?? 0;
+				return obj?.Target?.GetHashCode() ?? 0;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/DependencyObjectStore.cs]: Relevant to this PR? Should be a separate commit at least

**Comment by davidjohnoliver** [comment]
You should update https://github.com/nventive/Uno/blob/master/doc/articles/working-with-commandbar.md

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/DependencyObjectStore.cs]: Right, It's not relevant, I removed it and will do another PR for that.

**Comment by dr1rrb** [comment]
> You should update https://github.com/nventive/Uno/blob/master/doc/articles/working-with-commandbar.md

Right, I searched (a bit ;) ) for the doc but I didn't find it. We should have a convention to name all the files by the control name and avoid prefix or suffix like "working with".
